subprojects/language-native/src/main/java/org/gradle/language/cpp/CppApplication.java

```diff
@@ -34,12 +34,26 @@
     CppExecutable getDevelopmentBinary();
 
     /**
-     * Returns the debug executable for this application.
+     * Returns the dynamically linked debug executable for this application.
      */
     CppExecutable getDebugExecutable();
 
     /**
-     * Returns the release executable for this application.
+     * Returns the dynamically linked release executable for this application.
      */
     CppExecutable getReleaseExecutable();
+
+    /**
+     * Returns the statically linked debug executable for this application.
+     *
+     * @since 4.4
+     */
+    CppExecutable getDebugStaticExecutable();
```

Comment:
these don't make sense for applications, or at least, not many applications at all.